From 6b53de6cec92f282451a707175eb6eb083d34a23 Mon Sep 17 00:00:00 2001
From: Pratham Malaviya <prathamm@mechasystems.com>
Date: Mon, 22 Sep 2025 16:16:50 +0530
Subject: [PATCH] fix: LTE module detection issue

This patch adds hog configuration for GPIOs to
successfully detect the LTE module

Signed-off-by: Pratham Malaviya <prathamm@mechasystems.com>

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts b/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts
index 87e98e77bbca..f5a4e97bd5a5 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts
@@ -759,6 +759,30 @@ &pcie {
         status = "okay";
 };
 
+&gpio1 {
+
+	lte_reset {
+		gpio-hog;
+		gpios = <16 GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "LTE-RESET";
+	};
+
+	lte_pwr {
+		gpio-hog;
+		gpios = <17 GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "LTE-PWR";
+	};
+
+	lte-ap-mode {
+		gpio-hog;
+		gpios = <27 GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "LTE-AP-DIS";
+	};
+};
+
 &gpu_2d {
 	status = "okay";
 };
-- 
2.25.1

